### PR TITLE
Swappable Tools using Equipment Manager

### PR DESCRIPTION
**Pull request by wstampley**
POC to see what it might look like to manage swappable tools in equipment manager entries, since some of the best (fastest) forging tools are the swappable tongs.

@Dartellum - Seems like you've done most of the crafting scripts - thoughts on the approach?

YAML Config
```
############################# TOOLS
- :adjective: clockwork
  :name: tongs
  :is_leather: false
  :swappable: true
  :crafts: [forging]
  :tool_types: [tongs, shovel]
- :adjective: ball-peen
  :name: hammer
  :is_leather: false
  :tie_to: agonite.toolstrap
  :crafts: [forging]
  :tool_types: [hammer]

# I'm using a mix of gear entries and toolbelts to check for backwards compatibility, but you should be able to use all gear if you want
forging_belt:
  name: agonite toolstrap
  items: 
  - corrugated-hide bellows
  - stirring rod
  - complex stamp

# Blank out any tool entries for tools you don't have to keep Appraisal from getting confused
forging_tools:
- corrugated-hide bellows
- stirring rod
- complex stamp
- plain pliers # In backpack
tinkering_tools:
outfitting_tools:
alchemy_tools:
carving_tools:
enchanting_tools:
```
Updated / Validated Crafting Scripts:
- carve
- carve-bead
- craft
- forge
- mine
- mining-buddy
- sew
- shape
- smelt
- smelt-deeds
- smith
- workorders

Un-updated Scripts:
- alchemy
- arrows
- bolts
- clean-leather
- clean-lumber
- enchant
- spin
- tinker
- weave-cloth

### CODE REVIEW

File: equipmanager.lic [hunk 1]
```diff
@@ -18,7 +26,32 @@ class EquipmentManager
...
+    return swap_to_tool_type?(item, tool_type)
+  end
+
+  def swap_to_tool_type?(item, tool_type)
+    tool_swap_command = "adjust my #{item.adjective} #{item.name}"
+
+    item.tool_types.length.times {
+      case DRC.bput(tool_swap_command, SHOVEL_MATCHES, TONGS_MATCHES)
+      when *SHOVEL_MATCHES
+        return true if tool_type =~ /shovel/i
```
Comment (KatoakDR) [inline equipmanager.lic:49]: If the output matches the “when” statement, are the regex matches on tool_type for shovel or tongs still necessary? Could a tool have the right message but different noun?

Comment (wstampley) [reply equipmanager.lic:49]: it's checking that the swap was to the tool_type you want. the tongs are both tongs and shovels, for example, so you have to swap them around till you get to what you want.

Those don't have to be regex checks against the tool_type param, they could be equality checks, etc.

**Comment by Dartellum** [comment]
Years ago I attempted to add the adjustable tongs. I had it working, sort of, but ran into something that made me put it on the back burner. I have not really thought about since. I think it was mining. The tongs I have, in storage somewhere, are the shovel and tong combo.

**Comment by MahtraDR** [comment]
Hi @wstampley Are you planning on coming back to this by chance, or shall we close it? Thanks!

**Comment by wstampley** [comment]
> Hi @wstampley Are you planning on coming back to this by chance, or shall we close it? Thanks!

@MahtraDR - I've stopped playing and don't really intend to come back to this PR.

The branch was in a working state for most of the crafts. The methods in equipment manager work well, and it's mostly just the process of switching to them in the craft scripts I didn't finish and exhaustive testing that I didn't finish.

If no one wants to take over the work, feel free to close this. If someone does want to take it over, I'm happy to walk them through the approach and current state of things.

**Comment by MahtraDR** [comment]
> > Hi @wstampley Are you planning on coming back to this by chance, or shall we close it? Thanks!
> 
> 
> 
> @MahtraDR - I've stopped playing and don't really intend to come back to this PR.
> 
> 
> 
> The branch was in a working state for most of the crafts. The methods in equipment manager work well, and it's mostly just the process of switching to them in the craft scripts I didn't finish and exhaustive testing that I didn't finish.
> 
> 
> 
> If no one wants to take over the work, feel free to close this. If someone does want to take it over, I'm happy to walk them through the approach and current state of things.

Thanks for responding mate. 

All good. We'll do a review of this, and see what else there is to do and if we keep it or not. 

Thanks, and good luck!

**Comment by asechrest** [comment]
A lot of work was put into this but it doesn't look like anyone is willing to take up the banner and complete it. Thanks for the work on it. Closing due to inactivity. We can continue to reference it for ideas as we built out the use of the swappable tools.

**Pull request closed without merging**